README.md

```diff
@@ -72,6 +72,12 @@ $$
   FROM pg_catalog.pg_stats;
 $$ LANGUAGE sql VOLATILE SECURITY DEFINER;
 
+CREATE OR REPLACE FUNCTION pganalyze.get_relation_stats_ext() RETURNS SETOF pg_stats_ext AS
+$$
+  /* pganalyze-collector */ SELECT row_to_json(se.*)::jsonb - 'most_common_vals'
+  FROM pg_catalog.pg_stats_ext se;
+$$ LANGUAGE sql VOLATILE SECURITY DEFINER;
```

Comment:
[note for reviewer] We want to introduce this helper function to grab the stats data, but ensure that we won't grab the sensitive data. `most_common_vals` could include the actual values of the table, so we want to make sure excluding this. We do the similar thing above, specifying `NULL::anyarray` for `pg_stats` for the column like `most_common_vals` or `histogram_bounds`.
However, we can't do the similar way as `pg_stats` above because the columns of `pg_stats_ext` differs depending on the Postgres version (from 15, `inherited` column is introduced). As a workaround, we grab columns as jsonb except `most_common_vals`.

Reply:
Separately, this could be marked `STABLE` instead of `VOLATILE`, no? https://www.postgresql.org/docs/current/sql-createfunction.html

Reply:
> Separately, this could be marked `STABLE` instead of `VOLATILE`, no?

Good point and I agree, I actually thought about this while creating this, though I went with `VOLATILE` as the other functions that we define here are all `VOLATILE`. However, I think all of these functions actually can be `STABLE`. Should we update them all?

> we could start querying sensitive data if a column similar to `most_common_vals` is introduced in a future Postgres version

I think this is definitely a valid point, though if I can wear a lazy(?) hat, allowlisting (your version) instead of having a blocklist (current PR) will make it harder to use any "good" new columns introduced as the user will need to update the function to support it too. For example, above `get_column_stats()` is using the allowlisting with `pg_catalog.pg_stats`. This view hasn't changed for a long time (last time it changed was 9.2). On the other hand, `pg_stats_ext` has changed twice recently, both adding useful columns as stats data and we actually collect these data with this PR. It's possible that some columns will be introduced in the future releases that we want to start collecting metrics from.
That said, this function is indeed focusing more on the security perspective, so being convenience (of the maintainer) over the security sounds wrong, so I'd say that we should go with Maciek's version.

Just wanted to loop @lfittl too to get +1 from him too.

Reply:
How about we write it like this, for any fields that are not on all versions? (i.e. `exprs` and `inherited`)

```sql
SELECT statistics_schemaname, statistics_name, dependencies, n_distinct, most_common_val_nulls, most_common_freqs, most_common_base_freqs,
       row_to_json(se.*)::jsonb ->> 'inherited' AS inherited,
       row_to_json(se.*)::jsonb ->> 'exprs' AS exprs
  FROM pg_catalog.pg_stats_ext se
```

(note I intentionally included n_distinct and the most_common_* fields except the text one, since they may prove useful in the future, and we already consider those safe in the context of pg_stats)

Reply:
Works for me.

Reply:
Uhmm actually this will return the non defined table/type, so we will need to use `RETURNS TABLE` with a function definition which will make the function even longer 🤔

Reply:
I'll push the update too but here is what I got:

```sql
CREATE OR REPLACE FUNCTION pganalyze.get_relation_stats_ext() RETURNS TABLE(
  statistics_schemaname text, statistics_name text,
  inherited boolean, n_distinct pg_ndistinct, dependencies pg_dependencies,
  most_common_val_nulls boolean[], most_common_freqs float8[], most_common_base_freqs float8[]
) AS
$$
  /* pganalyze-collector */ SELECT statistics_schemaname, statistics_name,
  (row_to_json(se.*)::jsonb ->> 'inherited')::boolean AS inherited, n_distinct, dependencies, 
  most_common_val_nulls, most_common_freqs, most_common_base_freqs
  FROM pg_catalog.pg_stats_ext se;
$$ LANGUAGE sql VOLATILE SECURITY DEFINER;
```

Reply:
> Good point and I agree, I actually thought about this while creating this, though I went with VOLATILE as the other functions that we define here are all VOLATILE. However, I think all of these functions actually can be STABLE. Should we update them all?

I was trying to recall why we mark these as volatile, but came up empty - I thought that its maybe related to them being SECURITY DEFINER, but reviewing the Postgres documentation there doesn't appear to be any reason why a security definer function needs to be marked as volatile.

Reply:
Ah yeah thanks for following up re: VOLATILE and STABLE. I think it's still the best to double check if changing this to `STABLE` works without problems, but I'd say that let's address this separately. I also have been thinking if we should update our guided setup script, I think they don't include lots of functions defined here. Well, let's also address that separately.